crates/primitives/src/net.rs

```diff
@@ -0,0 +1,335 @@
+use crate::PeerId;
+use bytes::{Buf, BufMut};
+use reth_rlp::{Decodable, DecodeError, Encodable, Header};
+use reth_rlp_derive::RlpEncodable;
+use secp256k1::{SecretKey, SECP256K1};
+use serde_with::{DeserializeFromStr, SerializeDisplay};
+use std::{
+    fmt,
+    fmt::Write,
+    net::{IpAddr, Ipv4Addr, Ipv6Addr, SocketAddr},
+    num::ParseIntError,
+    str::FromStr,
+};
+use url::{Host, Url};
+
+/// IpAddr octets
+#[derive(Debug, Copy, Clone)]
+pub enum Octets {
```

Comment:
these can be private?
please move type below NodeRecord, so NodeRecord is first type in this file

Reply:
Is there a reason we are using `Octets` instead of `std::net::IpAddr` with `Encodable`/`Decodable` support?

Reply:
good q, would be possible, with another feature.

feel free to tackle this in a follow-up!